src/meta/service/src/stream_service.rs

```diff
@@ -420,4 +421,71 @@ impl StreamManagerService for StreamServiceImpl {
             .await;
         Ok(Response::new(RecoverResponse {}))
     }
+
+    async fn list_actor_splits(
+        &self,
+        _request: Request<ListActorSplitsRequest>,
+    ) -> Result<Response<ListActorSplitsResponse>, Status> {
+        match &self.metadata_manager {
+            MetadataManager::V1(_) => Ok(Response::new(ListActorSplitsResponse {
+                // TODO: remove this when v1 is removed
+                actor_splits: vec![],
+            })),
+            MetadataManager::V2(mgr) => {
+                let SourceManagerRunningInfo {
+                    source_fragments,
+                    backfill_fragments,
+                    mut actor_splits,
+                } = self.stream_manager.source_manager.get_running_info().await;
+
+                let fragment_to_source: HashMap<_, _> =
+                    source_fragments
+                        .into_iter()
+                        .flat_map(|(source_id, fragment_ids)| {
+                            fragment_ids
+                                .into_iter()
+                                .map(move |fragment_id| (fragment_id, source_id))
+                        })
+                        .chain(backfill_fragments.into_iter().flat_map(
+                            |(source_id, fragment_ids)| {
+                                fragment_ids.into_iter().flat_map(
+                                    move |(fragment_id, upstream_fragment_id)| {
+                                        [
+                                            (fragment_id, source_id),
+                                            (upstream_fragment_id, source_id),
+                                        ]
+                                    },
+                                )
+                            },
+                        ))
+                        .collect();
+
+                let source_actors = mgr.catalog_controller.list_source_actors().await?;
+
+                let actor_splits = source_actors
+                    .into_iter()
+                    .flat_map(|(actor_id, fragment_id)| {
+                        let source_id = fragment_to_source
+                            .get(&(fragment_id as _))
+                            .copied()
+                            .map(|id| id as _)
+                            .unwrap_or_default();
+
+                        actor_splits
+                            .remove(&(actor_id as _))
+                            .unwrap_or_default()
+                            .into_iter()
+                            .map(move |split| list_actor_splits_response::ActorSplit {
```

Comment:
Can we add a column `type` to show whether it's source fragment or backfill fragment here?